utils/oscap_docker_python/get_cve_input.py

```diff
@@ -28,52 +29,106 @@ class getInputCVE(object):
     '''
 
     hdr = {'User-agent': 'Mozilla/5.0'}
+    hdr2 = [('User-agent', 'Mozilla/5.0')]
     url = "http://www.redhat.com/security/data/oval/"
     dist_cve_name = "Red_Hat_Enterprise_Linux_{0}.xml"
     dists = [5, 6, 7]
+    remote_pattern = '%a, %d %b %Y %H:%M:%S %Z'
 
-    def __init__(self, fs_dest):
+    def __init__(self, fs_dest, DEBUG=False):
         ''' Simple init declaration '''
         self.dest = fs_dest
+        self.DEBUG = DEBUG
 
     def _fetch_single(self, dist):
         '''
         Given a distribution number (i.e. 7), it will fetch the
-        distribution specific data file
+        distribution specific data file if upstream has a newer
+        input file.  Returns the path of file.
         '''
         cve_file = self.dist_cve_name.format(dist)
+        dest_file = join(self.dest, cve_file)
         dist_url = (urlparse.urljoin(self.url, cve_file))
+        if self._is_cache_same(dest_file, dist_url):
+            return dest_file
+
         _url = urllib2.Request(dist_url, headers=self.hdr)
+        # TODO
+        # When dist specific files are available in bz form, some
+        # of this logic may need to change
         try:
             resp = urllib2.urlopen(_url)
+
         except Exception as url_error:
-            raise Exception("Unable to fetch CVE inputs")
+            raise Exception("Unable to fetch CVE inputs due to"
+                            .format(url_error))
 
-        # TODO
-        # When dist specific files are available in bz form,
-        # will need to unbz these
-        fh = open(join(self.dest, cve_file), "w")
+        fh = open(dest_file, "w")
         fh.write(resp.read())
         fh.close()
 
-    def _is_recent_enough(self, hours, dist):
+        # Correct Last-Modified timestamp
+        remote_ts = dict(resp.info())['last-modified']
+        epoch = datetime.datetime.utcfromtimestamp(0)
+        remote_dt = datetime.datetime.strptime(remote_ts, self.remote_pattern)
+        seconds_epoch = (remote_dt - epoch).total_seconds()
+        utime(dest_file, (seconds_epoch, seconds_epoch))
+
+        return self.dest
```

Comment:
Return value is currently used only in "fetch_dist_data" function, which is not called from other part of code now. I think, it should be "return dest_file", or am I wrong?